src/jsTyping/jsTyping.ts

```diff
@@ -293,8 +293,13 @@ namespace ts.JsTyping {
                 const ownTypes = packageJson.types || packageJson.typings;
                 if (ownTypes) {
                     const absolutePath = getNormalizedAbsolutePath(ownTypes, getDirectoryPath(normalizedFileName));

```

Comment:
I think _this_ line is a problem. A `"types"` field of `"index"` is _supposed_ to resolve to `index.d.ts` or `index.ts` in that priority, at least under `commonjs` module resolution.

Reply:
Huh, not seeing any attempt to do that, I just assumed we were keeping it simple.  Are the rules for `types` and `typings` different?

Reply:
It looks like DT [packages](https://unpkg.com/browse/source-map@0.7.0/package.json) are published with `"types": "index.d.ts"`

Reply:
@andrewbranch Do you want to call `resolveModuleName` as [here](https://github.com/microsoft/TypeScript/blob/9766757ee6b4df479330c170bec3a8951dd95744/src/typingsInstallerCore/typingsInstaller.ts#L22)?  Obviously, that's a much bigger change, but it seems reasonable.

Reply:
Done.

Reply:
I'm not familiar with this code, so I don't know how important perf is. DT packages use `.d.ts` now but they used a bar `"index"` for a long time, packages that haven't seen updates in the last year or two still have that. It seems reasonable to use some cheap guesses if perf is much more important than correctness here.

Reply:
It's happening asynchronously out of proc, so I don't think perf is essential.

Any idea how DT used to work?  I don't think this code has changed in a while.

Reply:
@andrewbranch Are you sure about the way DT used to work?  The TI has used the absolute path for [years](https://github.com/microsoft/TypeScript/commit/284d9f527c4162e3d044f93a5bfa3e1b1940fba2) (though maybe something downstream fixed up the paths?).

Reply:
It also looks like we don't do module resolution in [other places](https://github.com/microsoft/TypeScript/blob/0163a624b7d91a29cf24b4d411fd6d52fb5339cb/src/compiler/moduleSpecifiers.ts#L735).

Reply:
Reverting to dropping extensionless paths.

Reply:
@amcasey FYI that code you linked is (a) new and (b) sometimes part of a very hot path where we want to probe the FS as little as possible.